### ISSUE TITLE
keycloak_ldap_mapper with user-attribute-ldap-mapper

### ISSUE THREAD

**Issue by yenda23** [treydock/puppet-module-keycloak#44]
I'm trying to remap first name attribute from 'cn' to 'firstName', but with no luck, puppet code looks like this:
```
    keycloak_ldap_mapper { "first name for LDAP-${realm_name} on ${realm_name}":
      ensure               => 'present',                     
      type                 => 'user-attribute-ldap-mapper',  
      resource_name        => 'first name',                  
      user_model_attribute => 'firstName',                   
      ldap_attribute       => 'givenName',                   
    }   
```
and json request like this:
```
{
  "providerId": "user-attribute-ldap-mapper",
  "providerType": "org.keycloak.storage.ldap.mappers.LDAPStorageMapper",
  "config": {
    "ldap.attribute": [
      "givenName"
    ],
    "is.mandatory.in.ldap": [
      "false"
    ],
    "write.only": [
      "false"
    ]
  }
}
```
As I looked into json request made by this function but with full-name-ldap-mapper type (as you've described in doc - and this is working), it seems like json is missing some variables.
Is it bug or I'm doing somethin wrong?

**Comment by treydock** [comment]
I will see if I can reproduce but it looks like the generated JSON is missing `user.model.attribute` in the `config` block.

**Comment by yenda23** [comment]
@treydock this code produces an error everytime I run it.

**Comment by treydock** [comment]
@yenda23 #46 merged to master should fix the issue.  I think what was happening was wrong `id` was being used when trying to modify the ldap mapper that already existed in Keycloak.

**Comment by treydock** [comment]
Fix released in 3.3.0
